angular-dfp.js

```diff
@@ -429,10 +427,18 @@ angular.module('ngDfp', [])
                 return;
               }
 
-              $timeout(function () {
+              timeoutPromise = $timeout(function () {
                 DoubleClick.refreshAds(id);
               }, scope.timeout);
             });
+
+            // Cancel $interval and $timeout service when DOM destroy
+            element.on('$destroy', function() {
```

Comment:
Yeah I feel like `element.on` is more of a propagation of the angular `$destroy` event on the DOM element itself, so `scope.$on` feels more correct.